src/Message/MessageHandlerInterface.php

```diff
@@ -47,6 +47,14 @@ public function onBinary(string $data, AbstractConnection $connection);
      * This callback is call when there is an error on the websocket protocol communication.
      *
      * @param WebsocketException $e
+     * @param AbstractConnection $connection
      */
     public function onError(WebsocketException $e, AbstractConnection $connection);
+
+    /**
+     * Is called when the connection is closed by the client
+     *
+     * @param AbstractConnection $connection
+     */
+    public function onDisconnect(AbstractConnection $connection);
```

Comment:
An other solution would be creating an other interface, like `SupportsDisconnectionInterface`, which defines this method, so you can add the handler only if this interface is implemented.

\+ : No BC break, no abstracted method, and more visible/explicit imo
\- : overengineering?

There's my 5 cents, but the current implementation is fine to me.

Reply:
Yes it would be an option. But this feature is so basic that I really think it's ok to enforce this new method. (also, is that so much people using woketo at this time ? :p)